src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -653,47 +663,34 @@ def resolve_analyzed_atomic_cell(
         symmetry_analyzer: 'SymmetryAnalyzer',
         cell_type: str,
         logger: 'BoundLogger',
-    ) -> 'Optional[AtomicCell]':
+    ) -> 'Optional[Cell]':
         """
-        Resolves the `AtomicCell` section from the `SymmetryAnalyzer` object and the cell_type
-        (primitive or conventional).
-
-        Args:
-            symmetry_analyzer (SymmetryAnalyzer): The `SymmetryAnalyzer` object used to resolve.
-            cell_type (str): The type of cell to resolve, either 'primitive' or 'conventional'.
-            logger (BoundLogger): The logger to log messages.
-
-        Returns:
-            (Optional[AtomicCell]): The resolved `AtomicCell` section or None if the cell_type
-            is not recognized.
+        Resolves an atomic cell from the symmetry_analyzer data.
+        Note: In the new design, the cell contains only geometric data.
+        The per-atom electronic/state information is stored at the ModelSystem level (atom_states).
         """
-        if cell_type not in ['primitive', 'conventional']:
-            logger.error(
-                "Cell type not recognized, only 'primitive' and 'conventional' are allowed."
-            )
+        try:
+            wyckoff = symmetry_analyzer.get_wyckoff_letters(cell_type)
+            equivalent_atoms = getattr(
```

Comment:
this is not very pythonic, i suggest having a map of cell type and get system function